vpr/src/place/place_constraints.h

```diff
@@ -65,6 +65,25 @@ void propagate_place_constraints();
 
 void print_macro_constraint_error(const t_pl_macro& pl_macro);
 
+inline bool floorplan_legal(const t_pl_atom_blocks_to_be_moved& blocks_affected) {
+    bool floorplan_legal;
+
+    for (int i = 0; i < blocks_affected.num_moved_blocks; i++) {
```

Comment:
If this is a highly critical function, it may be worth making the upper bound a constant by creating a num_moved_blocks variable and using that as the upper bound.